### ISSUE TITLE
Error: W tensorflow/core/util/ctc/ctc_loss_calculator.cc:144] No valid path found.

### ISSUE THREAD

**Issue by bluseking** [MaybeShewill-CV/CRNN_Tensorflow#122]
NFO: 08-20 17:34:54: train_shadownet.py:148 * 140217671526144 Epoch: 542 cost= 69.799927 seq distance=  0.910998 train accuracy=  0.009455
INFO: 08-20 17:34:54: train_shadownet.py:148 * 140217671526144 Epoch: 543 cost= 69.512672 seq distance=  0.916690 train accuracy=  0.011318
INFO: 08-20 17:34:55: train_shadownet.py:148 * 140217671526144 Epoch: 544 cost= 69.189919 seq distance=  0.917188 train accuracy=  0.008049
2018-08-20 17:34:55.936201: W tensorflow/core/util/ctc/ctc_loss_calculator.cc:144] No valid path found.
INFO: 08-20 17:34:56: train_shadownet.py:148 * 140217671526144 Epoch: 545 cost=       inf seq distance=  0.925914 train accuracy=  0.006089
INFO: 08-20 17:34:56: train_shadownet.py:148 * 140217671526144 Epoch: 546 cost= 72.633850 seq distance=  0.903907 train accuracy=  0.014168
INFO: 08-20 17:34:57: train_shadownet.py:148 * 140217671526144 Epoch: 547 cost= 70.695236 seq distance=  0.903769 train accuracy=  0.012890
INFO: 08-20 17:34:57: train_shadownet.py:148 * 140217671526144 Epoch: 548 cost= 73.307205 seq distance=  0.915209 train accuracy=  0.007880
INFO: 08-20 17:34:58: train_shadownet.py:148 * 140217671526144 Epoch: 549 cost= 68.913414 seq distance=  0.897843 train accuracy=  0.006439
INFO: 08-20 17:34:59: train_shadownet.py:148 * 140217671526144 Epoch: 550 cost= 70.345169 seq distance=  0.942483 train accuracy=  0.003483
2018-08-20 17:34:59.575925: W tensorflow/core/util/ctc/ctc_loss_calculator.cc:144] No valid path found.
INFO: 08-20 17:34:59: train_shadownet.py:148 * 140217671526144 Epoch: 551 cost=       inf seq distance=  0.909413 train accuracy=  0.010943
INFO: 08-20 17:35:00: train_shadownet.py:148 * 140217671526144 Epoch: 552 cost= 69.603500 seq distance=  0.919080 train accuracy=  0.001563
INFO: 08-20 17:35:00: train_shadownet.py:148 * 140217671526144 Epoch: 553 cost= 71.181335 seq distance=  0.916861 train accuracy=  0.008120
INFO: 08-20 17:35:01: train_shadownet.py:148 * 140217671526144 Epoch: 554 cost= 70.934387 seq distance=  0.915586 train accuracy=  0.012421
INFO: 08-20 17:35:02: train_shadownet.py:148 * 140217671526144 Epoch: 555 cost= 72.648407 seq distance=  0.913969 train accuracy=  0.005334
INFO: 08-20 17:35:02: train_shadownet.py:148 * 140217671526144 Epoch: 556 cost= 68.141663 seq distance=  0.887581 train accuracy=  0.007994
INFO: 08-20 17:35:03: train_shadownet.py:148 * 140217671526144 Epoch: 557 cost= 69.755150 seq distance=  0.934300 train accuracy=  0.001563
INFO: 08-20 17:35:04: train_shadownet.py:148 * 140217671526144 Epoch: 558 cost= 70.684875 seq distance=  0.921868 train accuracy=  0.007672
INFO: 08-20 17:35:04: train_shadownet.py:148 * 140217671526144 Epoch: 559 cost= 71.249802 seq distance=  0.914272 train accuracy=  0.004695
INFO: 08-20 17:35:05: train_shadownet.py:148 * 140217671526144 Epoch: 560 cost= 68.226639 seq distance=  0.911951 train accuracy=  0.004852
INFO: 08-20 17:35:06: train_shadownet.py:148 * 140217671526144 Epoch: 561 cost= 71.468231 seq distance=  0.929269 train accuracy=  0.014256
INFO: 08-20 17:35:06: train_shadownet.py:148 * 140217671526144 Epoch: 562 cost= 67.099030 seq distance=  0.892195 train accuracy=  0.010294
INFO: 08-20 17:35:07: train_shadownet.py:148 * 140217671526144 Epoch: 563 cost= 67.613251 seq distance=  0.92019

when I train the model it shows  tensorflow/core/util/ctc/ctc_loss_calculator.cc:144] No valid path found.Each  image in my training dataset about 20 to 24 characters/numbers long

**Comment by Jackson-Kang** [comment]
You have to increase "__C.ARCH.SEQ_LENGTH" in global_configuration. Give enough time steps to CTC so that it can find the path for long range label.

**Comment by bluseking** [comment]
I have set the SEQ_LENGTH to 30in the config.py file .I still get the same error




| |
卓可强
邮箱：wind_cloud1222@126.com
|

签名由 网易邮箱大师 定制

On 08/30/2018 17:13, Jackson Kang wrote:

You have to increase "__C.ARCH.SEQ_LENGTH" in global_configuration. Give enough time steps to CTC so that it can find the path for long range label.

**Comment by bluseking** [comment]
Do l need to set other parameters?




| |
卓可强
邮箱：wind_cloud1222@126.com
|

签名由 网易邮箱大师 定制

On 09/03/2018 12:52, bluseking wrote:
I have set the SEQ_LENGTH to 30in the config.py file .I still get the same error




| |
卓可强
邮箱：wind_cloud1222@126.com
|

签名由 网易邮箱大师 定制

On 08/30/2018 17:13, Jackson Kang wrote:

You have to increase "__C.ARCH.SEQ_LENGTH" in global_configuration. Give enough time steps to CTC so that it can find the path for long range label.

**Comment by Jackson-Kang** [comment]
After checking papers and my codes, I found that I did some mistake..

Sorry, you can't edit sequence length because the feature sequence of CNN is 25.
I think it is caused by long labels which occurred by not finding paths to the end of label.

I changed SEQ_LENGTH to 50 and LSTM hidden cell to 512. Although I thought it will not work due to the number of sequences from CNN, but it works without any message that you wrote.  I found that tf.reshape after affine transformation made be enable this, but don't know the reason why it normally works.

**Comment by mdbenito** [comment]
In order to change the maximum sequence length to $L$ one needs to increase the input image width to $4 L$.

**Comment by Jackson-Kang** [comment]
Then I want to ask a question. In this code, width of an image is fixed, so I think the model can not distinguish characters properly. If an image has large width (more than 100), it must be hard to train the model. Is there any reason the width of an image is fixed?

**Comment by MaybeShewill-CV** [comment]
@bluseking The repo has been updated you may test again:)

**Comment by bluseking** [comment]
@MaybeShewill-CV  Thanks for your help
